packages/agreement-process/src/services/readModelService.ts

```diff
@@ -318,51 +319,108 @@ async function searchTenantsByName(
 }
 
 // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
-function getDelegateAgreementsFilters(producerIds: TenantId[] | undefined) {
-  return producerIds && producerIds.length > 0
-    ? [
+const getDelegationLookup = (kind: DelegationKind) => [
+  {
+    $lookup: {
+      from: "delegations",
+      let: {
+        eserviceId: "$data.eserviceId",
+        consumerId: "$data.consumerId",
+        producerId: "$data.producerId",
+      },
```

Comment:
These are fields from the Agreement but having them here without context could be misleading, maybe add a comment?